service/src/main/java/uk/gov/hmcts/reform/fpl/service/PrepareDirectionsForDataStoreService.java

```diff
@@ -0,0 +1,239 @@
+                                                            DirectionAssignee assignee) {
+        if (event == COMPLY_ON_BEHALF_SDO) {
+            return addCourtAssigneeAndDirectionId(id, response);
+        } else {
+            return addResponderAssigneeAndDirectionId(response, authorisation, assignee, id);
+        }
+    }
+
+    //TODO: if name of user complying for court is added then this can be merged with logic from method below.
+    private Element<DirectionResponse> addCourtAssigneeAndDirectionId(UUID id, Element<DirectionResponse> element) {
+        return ElementUtils.element(element.getId(), element.getValue().toBuilder()
+            .assignee(COURT)
+            .directionId(id)
+            .build());
+    }
+
+    private Element<DirectionResponse> addResponderAssigneeAndDirectionId(Element<DirectionResponse> response,
+                                                                          String authorisation,
+                                                                          DirectionAssignee assignee,
+                                                                          UUID id) {
+        return ElementUtils.element(response.getId(), response.getValue().toBuilder()
+            .assignee(assignee)
+            .responder(getUsername(response, authorisation))
+            .directionId(id)
+            .build());
+    }
+
+    private String getUsername(Element<DirectionResponse> element, String authorisation) {
+        if (isEmpty(element.getValue().getResponder())) {
+            return userDetailsService.getUserName(authorisation);
+        } else {
+            return element.getValue().getResponder();
+        }
+    }
+
+    /**
+     * Adds a direction response element to a direction where the direction id matches.
+     *
+     * @param responses  a list of direction response elements.
+     * @param directions a list of directions.
+     */
+    public void addResponseElementsToDirections(List<Element<DirectionResponse>> responses,
+                                                List<Element<Direction>> directions) {
+        directions.forEach(direction -> responses.stream()
+            .filter(response -> response.getValue().getDirectionId().equals(direction.getId()))
+            .forEach(response -> {
+                direction.getValue().getResponses().removeIf(x -> response.getId().equals(x.getId()));
+                direction.getValue().getResponses().add(response);
+            }));
+    }
+
+    /**
+     * Gets responses with populated directionId and assignee wherever a response is present within a direction.
+     *
+     * @param directionMap a map of directions where assignee is key and value is a list of directions.
+     * @return a list of responses with hidden variables added.
+     */
+    public List<DirectionResponse> getResponses(Map<DirectionAssignee, List<Element<Direction>>> directionMap) {
```

Comment:
As this is getter seems suspicious to be in place where we have got data persisting ones, maybe it should be in main one? Or the call should be extracted from controller? I.e. this bit confuses me a little:

```
        List<DirectionResponse> responses = prepareDirectionsForDataStoreService.getResponses(directionsMap);
        List<Element<Direction>> directionsToComplyWith = directionHelperService.getDirectionsToComplyWith(caseData);
```
why directions are in main helper and responses in another one? Can be confusing where to find stuff.

Reply:
hmm, I guess because getResponses is also adding hidden variables to the response, confusing method, shall look at refactoring